tests/integration/utils.py

```diff
@@ -122,21 +122,21 @@ def _calculate_files_checksums_in_dir(root_dir: str) -> Dict:
     files_checksums = {}
 
     for dir_, _, files in os.walk(root_dir):
+        rel_dir = PurePath(dir_).relative_to(root_dir)
         for file_name in files:
-            rel_dir = os.path.relpath(dir_, root_dir)
-            rel_file = os.path.join(rel_dir, file_name)
+            rel_file = rel_dir.joinpath(file_name)
             if "-external-gitcommit-" in file_name:
                 files_checksums[rel_file] = _get_git_commit_from_tarball(
-                    os.path.join(root_dir, rel_file)
+                    root_dir.joinpath(rel_file)
                 )
             else:
-                files_checksums[rel_file] = _calculate_sha256sum(os.path.join(root_dir, rel_file))
+                files_checksums[rel_file] = _calculate_sha256sum(str(root_dir.joinpath(rel_file)))
```

Comment:
IMO we shouldn't convert the Paths back to strings when not necessary

I'd say the `_calculate_sha256sum` function should be changed to take a Path param